### PR TITLE
make sure that we create a header for the encrypted private key

### LINKED ISSUE #15825: [enc2] Recovery key doesn't seem to work
### Steps to reproduce
1. Setup OC from scratch
2. Enable encryption and the default enc module
3. Log out then log in again as admin
4. Enable recovery key by setting a recovery password in the admin page
5. Create a user "user1" with password "test"
6. Login as "user1" and upload some files
7. Enable recovery key in the personal page
8. Login as admin
9. Go to the users page
10. Enter the recovery password
11. Change the password of "user1" to "xxx" (imagine the user had forgotten the password)
12. Login as "user1" with the password "xxx"
13. Access files

### Expected result
"user1" still has access to files.

### Actual result
After login, "user1" sees a notification about invalid private keys, asks the user to enter their password in the personal page. However, "user1" has forgotten the old password and cannot do that.

### Version
ownCloud 8.1 alpha2

@schiesbn @th3fallen @DeepDiver1975

### PR DESCRIPTION
**Pull request by schiessle**
make sure that we create a header for the encrypted private key

cc @PVince81 @DeepDiver1975 @th3fallen 

fix #15825

### CODE REVIEW

**Comment by PVince81** [comment]
Tested, works now :+1:

**Comment by schiesbn** [comment]
cc @th3fallen maybe you can also review it? Thanks!

File: apps/encryption/tests/hooks/UserHooksTest.php [hunk 1]
```diff
@@ -101,8 +102,18 @@ public function testSetPassphrase() {
 			->method('symmetricEncryptFileContent')
 			->willReturn(true);
 
+		$this->cryptMock->expects($this->any())
```
Comment (nickvergessen) [inline apps/encryption/tests/hooks/UserHooksTest.php]: `$this->atLeastOnce()` to make sure it gets called?

**Comment by nickvergessen** [comment]
:+1: but we could change the `expects()` as mentioned to make sure that the header method gets called

File: apps/encryption/tests/hooks/UserHooksTest.php (hunk 1 above)
Comment (schiesbn) [inline apps/encryption/tests/hooks/UserHooksTest.php]: could be added, at the other hand I check the parameter of setPrivateKey if it contains the header which will not be there if the method isn't called

File: apps/encryption/tests/hooks/UserHooksTest.php [hunk 2]
```diff
@@ -101,8 +102,18 @@ public function testSetPassphrase() {
...
+			->method('generateHeader')
+			->willReturn(Crypt::HEADER_START . ':Cipher:test:' . Crypt::HEADER_END);
+
 		$this->keyManagerMock->expects($this->exactly(4))
-			->method('setPrivateKey');
+			->method('setPrivateKey')
+			->willReturnCallback(function ($user, $key) {
+				$header = substr($key, 0, strlen(Crypt::HEADER_START));
```
Comment (nickvergessen) [inline apps/encryption/tests/hooks/UserHooksTest.php]: btw, there is `assertStringStartsWith` for this ;)

File: apps/encryption/tests/hooks/UserHooksTest.php (hunk 1 above)
Comment (nickvergessen) [inline apps/encryption/tests/hooks/UserHooksTest.php]: yeah sure, my +1 still stands, feel free to merge

File: apps/encryption/tests/hooks/UserHooksTest.php (hunk 2 above)
Comment (schiesbn) [inline apps/encryption/tests/hooks/UserHooksTest.php]: good to know for the next time :)

**Pull request merged**